packager/certcache/certcache.go

```diff
@@ -669,30 +750,64 @@ func (this *CertCache) updateCertIfNecessary() {
 		this.setCerts(certs)
 		return
 	}
-	if d >= time.Duration(CertRenewalInterval) {
+	if d >= time.Duration(certRenewalInterval) {
 		// Cert is still valid, don't do anything.
-	} else if d < time.Duration(CertRenewalInterval) {
-		// Cert is still valid, but we need to start process of requesting new cert.
-		log.Println("Warning: Current cert crossed threshold for renewal, attempting to renew.")
-		certs, err := this.certFetcher.FetchNewCert()
-		if err != nil {
-			log.Println("Error trying to fetch new certificates from CA: ", err)
-			return
+	} else if d < time.Duration(certRenewalInterval) {
+		this.renewedCertsMu.Lock()
+		defer this.renewedCertsMu.Unlock()
+
+		// Check if we already have a renewal cert waiting, fetch a new cert if not.
+		if this.renewedCerts == nil {
+			// Cert is still valid, but we need to start process of requesting new cert.
+			log.Println("Warning: Current cert crossed threshold for renewal, attempting to renew.")
+			certs, err := this.certFetcher.FetchNewCert()
+			if err != nil {
+				log.Println("Error trying to fetch new certificates from CA: ", err)
+				return
+			}
+			this.setNewCerts(certs)
+		} else {
+			var ocspUpdateAfter time.Time
+
+			ocsp, _, errorOcsp := this.readOCSP(true)
```

Comment:
It's Go style to capitalize the initialisms inside camel-case, e.g. errorOCSP and newOCSP. (But keep it lowercase if it's the first one, like ocspUpdateAfter.)

Reply:
Done.